### DISCUSSION TITLE
Occasionally writers side are stucked when Writers and Readers are matching each other with hundreds of topics

### DISCUSSION THREAD

**Discussion by chunyisong** [eProsima/Fast-DDS#3726] [Support]
### Is there an already existing issue for this?

- [X] I have searched the existing issues

### Expected behavior

 Writers and Readers should always match each other normally successfully or failedly with hundreds of topics with discovery server,and Occasionally PDP/EDP stucking should never happen.

### Current behavior

Occasionally readers or writers side are stucked when Writers and Readers are matching each other with hundreds of topics with discovery server using Tcpv4.The more topics lead to a higher probability of being stucked.

### Steps to reproduce

1. On the Writer side ,i create a participant,a publisher,200 writers( each writer for a different topic)
2. On the Reader side,i create a participant,a subscriber,200 readers( each reader suscribe one writer);I will create two processes for each  reader participant which means totally 400 readers connect 200 writers.
3. Start a discovery server
4. Start writers and publish random data
5. Then  **start two readers processes almost at same time**  ,writers side will match some readers and then sucked (readers and writers side listeners will no more called)

### Fast DDS version/commit

v2.10.1

### Platform/Architecture

Ubuntu Focal 20.04 amd64

### Transport layer

TCPv4

**Comment by JLBuenoLopez** [comment]
Thanks for your contribution @chunyisong 

I am moving this ticket to the support discussion forum according to [Fast DDS CONTRIBUTING guidelines](https://github.com/eProsima/Fast-DDS/blob/master/CONTRIBUTING.md). In order to analyze this issue several other causes have to be studied, as network architecture, bandwidth, etc. You might consider contacting [Fast DDS commercial support team](https://github.com/eProsima/Fast-DDS#getting-help) for options about how to tackle this issue.
